src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -308,47 +320,48 @@ class HuiMapCard extends LitElement implements LovelaceCard {
     return color;
   }
 
-  private _getEntities = memoizeOne(
-    (
-      states: HassEntities,
-      config: MapCardConfig,
-      configEntities?: MapEntityConfig[]
-    ) => {
-      if (!states || !config) {
-        return undefined;
+  private _getSourceEntities = memoizeOne(
+    (states: HassEntities, config?: MapCardConfig): string[] => {
+      if (!states || !config || !config.geo_location_sources) {
+        return [];
       }
 
-      const geoEntities: string[] = [];
-      if (config.geo_location_sources) {
-        // Calculate visible geo location sources
-        const includesAll = config.geo_location_sources.includes("all");
-        for (const stateObj of Object.values(states)) {
-          if (
-            computeDomain(stateObj.entity_id) === "geo_location" &&
-            (includesAll ||
-              config.geo_location_sources.includes(stateObj.attributes.source))
-          ) {
-            geoEntities.push(stateObj.entity_id);
-          }
+      const entities: string[] = [];
+      // Calculate visible geo location sources
+      const includesAll = config.geo_location_sources.includes("all");
+      for (const stateObj of Object.values(states)) {
+        if (
+          computeDomain(stateObj.entity_id) === "geo_location" &&
+          (includesAll ||
+            config.geo_location_sources.includes(stateObj.attributes.source))
+        ) {
+          entities.push(stateObj.entity_id);
         }
       }
-
-      return [
-        ...(configEntities || []).map((entityConf) => ({
-          entity_id: entityConf.entity,
-          color: this._getColor(entityConf.entity),
-          label_mode: entityConf.label_mode,
-          focus: entityConf.focus,
-          name: entityConf.name,
-        })),
-        ...geoEntities.map((entity) => ({
-          entity_id: entity,
-          color: this._getColor(entity),
-        })),
-      ];
+      return entities;
     }
   );
 
+  private _getEntities = memoizeOne(
+    (
+      states: HassEntities,
```

Comment:
Since `states` is still part of this memoized call, `getEntities` will still return a new array every time a state is changed. I don't think we should solve that is `shouldUpdate`, for example if darkmode changes, this will cause an update and an auto fit, and in the future that could become more problematic.

Reply:
Can you elaborate on how you think we should address this issue?
My approach was to refine the old logic [here](https://github.com/home-assistant/frontend/blob/e1dc73e992a620d580cd321f40ffa9ca44b8cc2a/src/panels/lovelace/cards/hui-map-card.ts#L203):
```
if (this._config?.geo_location_sources) {
  if (oldHass.states !== this.hass.states) {
    return true;
  }
}
```
with a comparison of the relevant entities (lines 200-220 in the new code), instead of any entity in the system.

I think that this is a better logic than the old one, regardless of additional improvements we can make in the future. Isn't it?

Reply:
`_getEntities` was changed to return the same object when there is no change.